llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -2277,28 +2277,40 @@ void AArch64AsmPrinter::LowerLOADgotAUTH(const MachineInstr &MI) {
   const MachineOperand &GAMO = MI.getOperand(1);
   assert(GAMO.getOffset() == 0);
 
-  MachineOperand GAHiOp(GAMO);
-  MachineOperand GALoOp(GAMO);
-  GAHiOp.addTargetFlag(AArch64II::MO_PAGE);
-  GALoOp.addTargetFlag(AArch64II::MO_PAGEOFF | AArch64II::MO_NC);
+  if (MI.getParent()->getParent()->getTarget().getCodeModel() ==
+      CodeModel::Tiny) {
```

Comment:
```suggestion
  if (MI.getMF()->getTarget().getCodeModel() == CodeModel::Tiny) {
```
Not sure if there's an easier way to get to the target from MI off the top of my head, but you can get the MachineFunction w/o going through `getParent`